Swiftfin tvOS/Views/SelectUserView/Components/ServerSelectionMenu.swift

```diff
@@ -43,33 +40,97 @@ extension SelectUserView {
         }
 
         var body: some View {
-            Button {
-                let parameters = SelectUserCoordinator.SelectServerParameters(
-                    selection: _serverSelection,
-                    viewModel: viewModel
-                )
+            Menu {
```

Comment:
I somewhat disagree that the menu layout should differ compared to iOS. In fact, just copying over the iOS implementation compiles but the order of the elements differ. At least right now, on iOS I do like that the `Add`, `Edit`, and `All Servers` buttons have bottom layout priority on iOS and the list would grow up with the additional servers.

<details>
<summary>iOS Menu</summary>

![IMG_6673](https://github.com/user-attachments/assets/7d9ea7a3-226d-47d5-aadf-ebd076764964)
</details>

Now, the fact that happens on iOS is currently just manual manipulation of where the `Menu` is and where the contents will be laid out. [`menuOrder`](https://developer.apple.com/documentation/swiftui/view/menuorder(_:)) has since been implemented and we can currently use that for tvOS to get the order of the menu elements that we would like. We will soon do that for iOS since it was implemented in iOS 16.

Reply:
I'm having some trouble with this. I can manually force this order where `AddServer` is at the bottom but the `Menu` defaults to focusing the top item first. So, if I had 50 servers, I would need to scroll through all of them to get to `AllServers`. I was hoping `Priority` would be helpful but I am getting:

<img width="434" alt="Screenshot 2024-12-17 at 15 05 09" src="https://github.com/user-attachments/assets/d6fd5132-9c03-43d0-b339-61bf6a747038" />

Reply:
This is the current version I have:

![image](https://github.com/user-attachments/assets/5931391d-dd7d-4367-902e-d88dce807339)

It's kind of the reverse of iOS's order. The focus in this screenshot matches where focus goes on first Menu selection

Reply:
If you use `menuOrder(.fixed)` then it will be laid out in the order as statically defined in the code. That means that it will require switching around the sections and buttons. At least given our design where the `Menu` is always on the bottom, we can guarantee that the menu will be presented upwards and get the ordering that we want. Although, I find it odd `priority` isn't offered on tvOS.

Reply:
This is where I'm at now. The order is good but I don't love that the focus starts at the top. I've tried a couple different `@FocusState` attempts but it doesn't seem to be working how I expect it to. Maybe it's just getting the `onSelect` for `Menu` isn't working how I think it should. I've tried `onSubmit` and `onTapGesture` to no avail.

Here's what this looks like with a lot of items:

https://github.com/user-attachments/assets/8ea5ec66-1e4c-48c1-a442-a8d5caf858ad

Reply:
Hm, you're right that it may be annoying that it starts at the top. However for this I don't anticipate many servers being added, especially on tvOS, and we can figure that out later.

Reply:
Sounds good! If you're happy with that layout, then this should be good to go!